### PR TITLE
Fixup direction == 'rtl'

### LINKED ISSUE #4499: Border radius is not getting updated correctly in RTL
### Environment

If you are using latest version:
1. `react-native -v`:
react-native-cli: 2.0.1
react-native: 0.60.6
2. `react-native run-windows --info`:
System:
    OS: Windows 10 10.0.18363
    CPU: (12) x64 Intel(R) Xeon(R) CPU E5-1650 v3 @ 3.50GHz
    Memory: 19.41 GB / 31.91 GB
  Binaries:
    Node: 10.15.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.21.1 - C:\Program Files\nodejs\yarn.CMD
    npm: 6.4.1 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    react: 16.8.6 => 16.8.6
    react-native: 0.60.6 => 0.60.6
    react-native-windows: 0.60.0-vnext.155 => 0.60.0-vnext.155
  Installed UWP SDKs:
    10.0.16299.0
    10.0.17134.0
    10.0.17749.0
    10.0.17763.0
    10.0.18362.0

Then, specify:

- Target Platform Version(s):
10.0.18362.0

- Target Device(s):
Desktop

- Visual Studio Version:
Visual Studio 2017

- Build Configuration:
Debug
### Steps to Reproduce

(Write your steps here:)

1. Create a View and give it a border radius style using one of the: borderTopStartRadius, borderBottomStartRadius, borderTopEndRadius, borderBottomEndRadius
2. Some time after the View has been mounted, provide a new style with updated values for the above.
3. Observe in LTR this works as expected.
4. Switch to RTL, relaunch the app
5. Notice that on mount the border radius is applied as expected (start on he right, end on the left)
6. After the style is updated, the radius gets applied as if it was LTR (start on the left, end on the right)
### Expected Behavior

I would expect the style to behave the same way that it did on mount.

### Actual Behavior

After the style is updated, the radius gets applied as if it was LTR (start on the left, end on the right)

LTR behaves as  expected:
LTRradius
RTL after update style gets applied incorrectly (switches sides), on remounting it does applies correctly:
RTLradius

### Reproducible Demo

[https://snack.expo.io/SLis5ZWht](https://snack.expo.io/SLis5ZWht)
Put this component in an application somewhere, and launch with an RTL language.

### LINKED ISSUE #4645: marginLeft/marginRight/marginStart/marginEnd is backwards when direction == RTL
These properties have the opposite effect of iOS in RTL:
marginLeft
marginRight
marginStart
marginEnd

Min Repro:

https://snack.expo.io/vKVqJmmvB

Run on iOS and notice the margin is applied to the right side.

On Windows, notice the margin is applied to the left side instead of the right side

### LINKED ISSUE #4646: RNTester RTLExample page is broken on Windows
In RNTester, go to the RTLExample page.

Notice:  Nothing happens when you click on any of the buttons that toggle between LTR and RTL.

I debugged this a bit.  There are multiple problems that prevent this page from working on Windows:
-There's a central function that toggles between rtl/ltr, this function only does anything on iOS.
-Even after fixing the above, direction never toggles.

This needs to be further debugged and fixed.  As it stands this page isn't useful on Windows.

### LINKED ISSUE #4668: Provide iOS-compatible implementation for direction property
While investigating #4499 I started playing around with the direction property and found it doesn't work correctly.

To see this in action, see:
https://snack.expo.io/JV!BvpeCj

If you run this on iOS, the buttons and the Picker will flow RTL
If you run this on react-native-windows, the buttons will flow LTR, but the Picker will flow RTL

The basic problem is in RNW's code that handles the direction property.  We let yoga do the layout in RTL, but then we set the FlowDirection property on the backing XAML element.  This flips the RN children back to LTR.  But a native component like Picker will still pick up the XAML RightToLeft.  So you end up with a weird mix of LTR and RTL.

This bug is tracking fixing this by providing a "correct" implementation for direction.  iOS does it correctly.  We should too.

### PR DESCRIPTION
**Pull request by kmelmon**
Fixes #4668 
Fixes #4646 
Fixes #4645 
Fixes #4499 

This change fixes multiple RTL related issues.  At the most basic level, our implementation of direction == 'rtl' was incorrect.  The problem here was that we would do two things:
1) Yoga would see the direction == 'rtl' and do layout in an RTL fashion.  This results in absolute positions in a "flipped" coordinate space, which we would then push into XAML.
2) We also mapped the direction property to the XAML FlowDirection property and set this on the backing XAML element.  When FlowDirection == RightToLeft, XAML flips everything.

These two cancel each other out in the most basic scenario (a container with children stacked horizontally), so you'd end up with an LTR layout instead of RTL.

Also, there was code in place which tried to manually flip certain properties (padding, margin, and border), but this code was also incorrect.  Most importantly, this code is unnecessary.  It appears there was just a misunderstanding about how XAML applies these properties - it already flips these properties when in an RTL mode.  I manually verified each of these properties is supposed to flip in RTL mode, and XAML does flip them in RTL.  Also, after debugging some scenarios, I realized this code was buggy.  It was manually querying the FlowDirection of the element having its properties set, which isn't reliable.  The FlowDirection property inherits down the tree, so at the time a XAML element is being created, it hasn't inherited the property yet.  But later, after it is in the tree, and XAML layout runs, it will.  So the result of querying for FlowDirection varies depending on when we update the properties.  Yikes!  This caused the symptoms we saw in bug #4499, where the corner radius was sometimes flipped, depending on exactly when the property was applied.

This change fixes both sets of issues by doing the following:
1) Do not let yoga layout anything in RTL.  Always run yoga layout in LTR.
2) Push FlowDirection into XAML, and let XAML be the one and only place we flip into RTL.  This will do the correct thing for flexbox layout, and also allow native controls like ComboBox to pick up the correct FlowDirection as yoga doesn't do the layout of these controls, XAML does.

As a result of the above, I was able to also remove the code that was trying to manually flip certain properties as it's totally unnecessary.

Also, I noticed the RTLExample RNTester page essentially does nothing on Windows, as the page only actually applies direction == 'rtl' if the Platform is iOS.  This is because rtl isn't supported on Android.  The fix was simple - just check for "not Android".

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4683)

### CODE REVIEW

**Comment by kmelmon** [comment]
@kaiguo thanks.  Yes this fix is still needed.  Since we're still pushing FlowDirection == RightToLeft into the panel it would mirror the image without your change.

File: vnext/src/RNTester/js/examples/RTL/RTLExample.windows.js [hunk 1]
```diff
@@ -0,0 +1,749 @@
+/**
```
Comment (NickGerleman) [inline vnext/src/RNTester/js/examples/RTL/RTLExample.windows.js:1]: Could you mark the Windows specific changes with "// [windows" etc?

File: vnext/src/overrides.json [hunk 2]
```diff
@@ -814,6 +814,14 @@
       "type": "platform",
       "file": "RNTester\\js\\examples-win\\WindowsBrush\\WindowsBrushExample.windows.tsx"
     },
+    {
+      "type": "patch",
+      "file": "RNTester\\js\\examples\\RTL\\RTLExample.windows.js",
+      "baseFile": "RNTester\\js\\examples\\RTL\\RTLExample.js",
+      "baseVersion": "0.62.2",
+      "baseHash": "70a65dc896406e7ad971f0a9a770b7ea30f43b9e",
+      "issue": 678
```
[the remark this answers is not in the document]
Comment (kmelmon) [reply vnext/src/overrides.json:823]: I think there's a bug in the script.  I typed in 4678 and it truncated or something.

File: vnext/src/RNTester/js/examples/RTL/RTLExample.windows.js (hunk 1 above)
Comment (kmelmon) [reply vnext/src/RNTester/js/examples/RTL/RTLExample.windows.js:1]: they already do have // [windows]

File: vnext/src/overrides.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (NickGerleman) [reply vnext/src/overrides.json]: I think the "prompts" package behavior is to clear the number after a delay if you type something new. I've had the same thing pop up before. I can check if there's an option or new release that fixes the issue.

**Pull request merged**